### PR TITLE
Add lazy loading iframe docs

### PR DESCRIPTION
**Pull request by jazzsequence**
Closes humanmade/altis-documentation#201

Adds some docs for iframe lazy loading into the lazy loading documentation within the Media module.

### CODE REVIEW

File: docs/lazy-loading.md [hunk 1]
```diff
@@ -45,3 +45,31 @@ The radius needs to be tuned to each size individually. Facebook uses about 200
 Gaussholder includes a CLI command to help you tune the radius: pick a representative attachment or image file and use `wp gaussholder check-size <id_or_image> <radius>`. Adjust the radius until you get to roughly 200B, then check against other attachments to ensure they're in the ballpark.
 
 Note: changing the radius requires regenerating the placeholder data. Run `wp gaussholder process-all --regenerate` after changing radii or adding new sizes.
+
+## Lazy Loading iFrames
+
+iFrames are lazy-loaded by default using the [browser-level `loading` attribute](https://html.spec.whatwg.org/multipage/iframe-embed-object.html#attr-iframe-loading). Only `iframe` tags with both `width` and `height` attributes present will be lazy-loaded to avoid a negative impact on layout shifting. oEmbeded iframe content within `the_content`, `the_excerpt` and `widget_text_content` will have the `loading="lazy"` tag added by default where the web service has provided a `width` and `height` attribute.
```
Comment (roborourke) [inline docs/lazy-loading.md:51]:
```suggestion
Inline frames are lazy-loaded by default using the [browser-level `loading` attribute](https://html.spec.whatwg.org/multipage/iframe-embed-object.html#attr-iframe-loading). Only `<iframe>` tags with both `width` and `height` attributes present will be lazy-loaded to avoid a negative impact on layout shifting. Embedded `<iframe>` tags provided via oEmbed in content run through `the_content`, `the_excerpt` or `widget_text_content` filters will have the `loading="lazy"` attribute added by default where the web service has provided a `width` and `height` attribute.
```

I haven't seen the `iFrame` capitalisation anywhere before but we should make it consistent. On MDN for example in plain english they're always referred to as inline frames or `<iframe>`. The "i" should at least be capitalised in case Apple decide to sue us 😛

**Review by roborourke — COMMENTED** [review]
Have a suggestion to clarify some bits of assumed knowledge in the 1st paragraph but a more general question regarding the capitalisation of "iFrame". I feel like we should follow MDN's lead there but if you're aware of this being commonly used I'm good.

Also could document the function `wp_iframe_tag_add_loading_attr()` for adding the attribute.

File: docs/lazy-loading.md (hunk 1 above)
Comment (jazzsequence) [reply docs/lazy-loading.md:51]: A quick search shows a bunch of articles about iframes with a capital F but none of those are from respected documentation sources (except Microsoft docs). W3Schools actually capitalizes the "I" which is blasphemy. 😂

Screen Shot 2021-04-15 at 10 30 14 AM

This might go back to my memory of HTML 2.0 days where eVerYthInG waS camElCasE. "Inline frames" and the actual tag are _much_ better. 👍

File: docs/lazy-loading.md [hunk 2]
```diff
@@ -45,3 +45,31 @@ The radius needs to be tuned to each size individually. Facebook uses about 200
...
+add_filter( 'wp_lazy_loading_enabled', function( $default, $tag_name, $context ) {
+	if ( $tag_name === 'iframe' && $context === 'the_content'  ) {
+		return false;
+	}
+
+	return $default;
+}, 10, 3 );
+```
+
+You can also use the `wp_iframe_tag_add_loading_attr` filter to customize a specific iFrame tag. For example, if you wanted to disable lazy-loading on iFrames from a specific provider, like YouTube, you could use the following code:
```
Comment (jazzsequence) [inline docs/lazy-loading.md:65]:
```suggestion
You can also use the `wp_iframe_tag_add_loading_attr` filter to customize a specific `<iframe>` tag. For example, if you wanted to disable lazy-loading on inline frames from a specific provider, like YouTube, you could use the following code:
```

**Comment by jazzsequence** [comment]
Thank you for saving me from myself 😅
Will hold off on adding docs for the `wp_iframe_tag_add_loading_attr` _function_ as I think actually using that function would be an edge case (the only context I can imagine using it is if you're manually adding an iframe into a template and wrapping it with that function to lazy load it).

File: docs/lazy-loading.md [hunk 3]
```diff
@@ -45,3 +45,31 @@ The radius needs to be tuned to each size individually. Facebook uses about 200
...
 
 Note: changing the radius requires regenerating the placeholder data. Run `wp gaussholder process-all --regenerate` after changing radii or adding new sizes.
+
+## Lazy Loading iFrames
+
+Inline frames are lazy-loaded by default using the [browser-level `loading` attribute](https://html.spec.whatwg.org/multipage/iframe-embed-object.html#attr-iframe-loading). Only `<iframe>` tags with both `width` and `height` attributes present will be lazy-loaded to avoid a negative impact on layout shifting. Embedded `<iframe>` tags provided via oEmbed in content run through `the_content`, `the_excerpt` or `widget_text_content` filters will have the `loading="lazy"` attribute added by default where the web service has provided a `width` and `height` attribute.
+
+You can customize whether and how iFrames are lazy loaded using the `wp_lazy_loading_enabled` filter. For example, to disable lazy-loading of iFrames from post content entirely, you could use the following code:
```
Comment (roborourke) [inline docs/lazy-loading.md:53]:
```suggestion
You can customize whether and how inline frames are lazy loaded using the `wp_lazy_loading_enabled` filter. For example, to disable lazy-loading of inline frames from post content entirely, you could use the following code:
```

File: docs/lazy-loading.md [hunk 4]
```diff
@@ -45,3 +45,31 @@ The radius needs to be tuned to each size individually. Facebook uses about 200
 Gaussholder includes a CLI command to help you tune the radius: pick a representative attachment or image file and use `wp gaussholder check-size <id_or_image> <radius>`. Adjust the radius until you get to roughly 200B, then check against other attachments to ensure they're in the ballpark.
 
 Note: changing the radius requires regenerating the placeholder data. Run `wp gaussholder process-all --regenerate` after changing radii or adding new sizes.
+
+## Lazy Loading iFrames
```
Comment (roborourke) [inline docs/lazy-loading.md:49]:
```suggestion
## Lazy Loading Inline Frames
```

**Review by roborourke — APPROVED** [review]
